homeassistant/components/reolink/button.py

```diff
@@ -235,3 +235,12 @@ def __init__(
     async def async_press(self) -> None:
         """Execute the button action."""
         await self.entity_description.method(self._host.api)
+
+
+class ReolinkAvailabilityHostButtonEntity(ReolinkHostButtonEntity):
```

Comment:
Would it be cleaner to add a 'available_fn' lambda in `ReolinkButtonEntityDescription`, so a single `ReolinkHostButtonEntity` and `BUTTON_ENTITIES` list can be used.

Reply:
I think you can just even do a 

```py
@property
def available(self) -> bool:
    return self.entity_description.key == "reboot" or super().available
```

Reply:
Thank you for the suggestions, indeed this is much cleaner.
I have now implemented this.